src/compiler/scanner.ts

```diff
@@ -2455,6 +2457,35 @@ export function createScanner(languageVersion: ScriptTarget,
         return scanJsxAttributeValue();
     }
 
+    /** TODO: might need to return WhitespaceTrivia if only whitespace was encountered? */
+    function scanBigJsDocToken(): JSDocSyntaxKind { // can be configurable to skip almost everything (except newline and backtick) if backticks is true
+        startPos = tokenPos = pos;
+        tokenFlags = TokenFlags.None;
+        if (pos >= end) {
+            return token = SyntaxKind.EndOfFileToken;
+        }
+
+        let ch = codePointAt(text, pos);
+        while (pos < end) {
+            if (ch !== CharacterCodes.lineFeed && ch !== CharacterCodes.at && ch !== CharacterCodes.backtick && ch !== CharacterCodes.openBrace) {
+                // TODO: We can also be smarter about openBrace, backtick and at by looking at a tiny amount of context
+                pos++;
+            }
+            else {
+                break;
+            }
+            ch = codePointAt(text, pos);
+        }
+        if (pos === tokenPos) {
+            return scanJsDocToken();
+        }
+        else {
+            // TODO: Make sure this is right (and in the right place)
+            tokenValue = text.substring(tokenPos, pos);
+        }
+        return token = SyntaxKind.Identifier;

```

Comment:
I know that this is currently a proof-of-concept, but you should create an internal `SyntaxKind.JSDocTextSpan` that is used **only** in the parser